src/tokens/TokenImport.tsx

```diff
@@ -27,57 +30,149 @@ import variables from 'src/styles/variables'
-import networkConfig from 'src/web3/networkConfig'
-import { isAddress } from 'viem'
+import Logger from 'src/utils/Logger'
+import { publicClient } from 'src/viem'
+import networkConfig, { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { Address, formatUnits, getContract, isAddress } from 'viem'
+
+const TAG = 'tokens/TokenImport'
 
 type Props = NativeStackScreenProps<StackParamList, Screens.TokenImport>
 
+const FETCH_TOKEN_DETAILS_TIMEOUT_MS = 5_000
+
+interface TokenDetails {
+  address: string
+  symbol: string
+  decimals: number
+  name: string
+  balance: bigint
+}
+
 export default function TokenImportScreen(_: Props) {
   const { t } = useTranslation()
   const dispatch = useDispatch()
 
-  const [isCompleteAddress, setIsCompleteAddress] = useState(false)
   const [tokenAddress, setTokenAddress] = useState('')
   const [tokenSymbol, setTokenSymbol] = useState('')
   const [error, setError] = useState<string | null>(null)
   const [networkId] = useState(networkConfig.defaultNetworkId)
+
+  const walletAddress = useSelector(walletAddressSelector)
   const supportedTokens = useSelector((state) => tokensByIdSelector(state, [networkId]))
 
-  const validateAddress = (address: string) => {
-    if (!address) return
-
-    if (isAddress(address)) {
-      setIsCompleteAddress(true)
-      // TODO(RET-891): if already imported, set state as AddressState.AlreadyImported
-      const tokenId = getTokenId(networkId, address.toLowerCase())
-      if (supportedTokens[tokenId]) {
-        setError(t('tokenImport.error.alreadySupported'))
-      } else {
-        setError(null)
-      }
-    } else {
+  const validateAddress = (address: string): address is Address => {
+    if (!address) return false
+
+    if (!isAddress(address)) {
       setError(t('tokenImport.error.invalidToken'))
+      return false
+    }
+
+    // TODO(RET-891): if already imported, set state as AddressState.AlreadyImported
+    const tokenId = getTokenId(networkId, address.toLowerCase())
+    if (supportedTokens[tokenId]) {
+      setError(t('tokenImport.error.alreadySupported'))
+      ValoraAnalytics.track(AssetsEvents.import_token_already_supported, {
+        networkId,
+        tokenId,
+        tokenAddress,
+      })
+      return false
     }
-  }
 
-  const handleAddressFocus = () => {
-    setIsCompleteAddress(false)
     setError(null)
+    return true
+  }
+
+  const fetchTokenDetails = async (
+    tokenAddress: Address,
+    walletAddress: Address
+  ): Promise<TokenDetails> => {
+    const contract = getContract({
+      abi: erc20.abi,
+      address: tokenAddress,
+      publicClient: publicClient[networkIdToNetwork[networkId]],
+    })
+
+    const [symbol, decimals, name, balance] = await Promise.all([
+      contract.read.symbol(),
+      contract.read.decimals(),
+      contract.read.name(),
+      contract.read.balanceOf([walletAddress]),
```

Comment:
Do we need to fetch the balance?

Reply:
Not for this screen, however, once imported we'll need to show the balance.

If it fails to read balance here, then we know it's not an ERC-20 token and avoid importing it.